quixstreams/platforms/quix/api.py

```diff
@@ -61,16 +66,40 @@ def request(self, method, url, **kwargs):
     @property
     def default_workspace_id(self) -> str:
         if not self._default_workspace_id:
-            raise self.UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
+            raise UndefinedQuixWorkspaceId(
+                f"A Quix Cloud Workspace ID is required; "
+                f"set with environment variable {QUIX_ENVIRONMENT.WORKSPACE_ID}"
+            )
         return self._default_workspace_id
 
     @default_workspace_id.setter
     def default_workspace_id(self, value):
         self._default_workspace_id = value
 
+    def _response_handler(self, r: requests.Response, *args, **kwargs):
+        """
+        Custom callback/hook that is called after receiving a request.Response
+
+        Catches non-200's and passes both the original exception and the Response body.
+
+        Note: *args and **kwargs expected for hook
+        """
+        try:
+            r.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            try:
+                reason_text = e.response.json()
+            except requests.exceptions.JSONDecodeError:
+                reason_text = e.response.text
+
+            reason_text = reason_text or "Invalid URL"
```

Comment:
Where does the `"Invalid URL"` come from?

Reply:
This is if you hit an endpoint that just doesn't exist period (there isn't an error message/body is empty).

I think it'll only come into play if you were to use the session object to try and hit other endpoints manually tbh (which is how I found it), but I didn't test all of them to know for sure...so I just added this as a way of clarifying it's a bad endpoint if no error was given.

Reply:
I think this is related to https://github.com/quixio/quix-streams/pull/320#discussion_r1549306870

If we catch only the particular exceptions, the situation you tried to handle will be eliminated (every exception will be guaranteed to have a response)

Reply:
Yeah, that's a diff mistake =) But what I was attempting to cover here is still applicable.

If you hit an endpoint that doesn't exist, the request has no response body (at least, for our API anyway). It still throws the error (like `HTTPError: 404 Client Error: blah blah`) That being said, IDK if this can happen with things like 403's as well, which "Invalid URL" would be inappropriate for.

I think I found something a little cleaner that shortens the error when we have more info on what failed, and otherwise just returns the original as-is if not.